maven/rules.bzl

```diff
@@ -88,12 +97,14 @@ def _generate_class_jar(ctx, pom_file):
     maven_coordinates = _parse_maven_coordinates(target[JarInfo].name)
 
     jar = None
-    if hasattr(target, "files") and target.files.to_list() and target.files.to_list()[0].extension == "jar":
+    if (_is_android_library(target)):
+        jar = target[AndroidLibraryAarInfo].aar
```

Comment:
The information we need here is only the path to the JAR/AAR file.

Is there any way we could retrieve this in a way that is generic to both Java JARs and Android AARs, without requiring a check for `_is_android_library`? Since ideally we would like to keep this rule as generalised as possible.